src/main/java/com/alipay/oceanbase/hbase/util/OHConnectionImpl.java

```diff
@@ -0,0 +1,173 @@
+        this.closed = false;
+    }
+
+    @Override
+    public Configuration getConfiguration() {
+        return this.conf;
+    }
+
+    private ExecutorService getBatchPool() {
+        if (batchPool == null) {
+            synchronized (this) {
+                if (batchPool == null) {
+                    this.batchPool = getThreadPool(
+                        conf.getInt("hbase.hconnection.threads.max", 256),
+                        conf.getInt("hbase.hconnection.threads.core", 256), "-shared-", null);
+                    this.cleanupPool = true;
+                }
+            }
+        }
+        return this.batchPool;
+    }
+
+    private ExecutorService getThreadPool(int maxThreads, int coreThreads, String nameHint,
+                                          BlockingQueue<Runnable> passedWorkQueue) {
+        // shared HTable thread executor not yet initialized
+        if (maxThreads == 0) {
+            maxThreads = Runtime.getRuntime().availableProcessors() * 8;
+        }
+        if (coreThreads == 0) {
+            coreThreads = Runtime.getRuntime().availableProcessors() * 8;
+        }
+        long keepAliveTime = conf.getLong("hbase.hconnection.threads.keepalivetime", 60);
+        BlockingQueue<Runnable> workQueue = passedWorkQueue;
+        if (workQueue == null) {
+            workQueue = new LinkedBlockingQueue<Runnable>(
+                maxThreads
+                        * conf.getInt(HConstants.HBASE_CLIENT_MAX_TOTAL_TASKS,
+                            HConstants.DEFAULT_HBASE_CLIENT_MAX_TOTAL_TASKS));
+        }
+        ThreadPoolExecutor tpe = new ThreadPoolExecutor(coreThreads, maxThreads, keepAliveTime,
+            TimeUnit.SECONDS, workQueue, Threads.newDaemonThreadFactory(toString() + nameHint));
+        tpe.allowCoreThreadTimeOut(true);
+        return tpe;
+    }
+
+    @Override
+    public HTableInterface getTable(TableName tableName) throws IOException {
+        return getTable(tableName, getBatchPool());
+    }
+
+    @Override
+    public HTableInterface getTable(TableName tableName, ExecutorService pool) throws IOException {
+        return new OHTable(conf, tableName.getName(), pool);
+    }
+
+    @Override
+    public BufferedMutator getBufferedMutator(TableName tableName) throws IOException {
+        return getBufferedMutator(new BufferedMutatorParams(tableName));
+    }
+
+    @Override
+    public BufferedMutator getBufferedMutator(BufferedMutatorParams params) throws IOException {
+        if (params.getTableName() == null) {
+            throw new IllegalArgumentException("TableName cannot be null.");
+        }
+        if (params.getPool() == null) {
+            params.pool(HTable.getDefaultExecutor(getConfiguration()));
+        }
+        if (params.getWriteBufferSize() == BUFFERED_PARAM_UNSET) {
+            params.writeBufferSize(connectionConfig.getWriteBufferSize());
+        }
+        if (params.getMaxKeyValueSize() == BUFFERED_PARAM_UNSET) {
```

Comment:
why not use BufferedMutatorParams.UNSET directly?

Reply:
private param